lib/fuse_lowlevel.c

```diff
@@ -1909,6 +1915,14 @@ static void do_init(fuse_req_t req, fuse_ino_t nodeid, const void *inarg)
 			se->conn.capable |= FUSE_CAP_HANDLE_KILLPRIV;
 		if (arg->flags & FUSE_NO_OPENDIR_SUPPORT)
 			se->conn.capable |= FUSE_CAP_NO_OPENDIR_SUPPORT;
+		if (!arg->flags & FUSE_MAX_PAGES) {
+			size_t max_bufsize =
+				FUSE_DEFAULT_MAX_PAGES_PER_REQ * getpagesize()
+				+ HEADER_SIZE;
+			if (bufsize > max_bufsize) {
```

Comment:
Why do we have to limit the bufsize only if FUSE_MAX_PAGES is not set? I'd think that we always need *some* limit.

Reply:
In fuse_session_new() the bufsize is initialized to the largest possible value. Only if we detect that FUSE_MAX_PAGES is not set we decrease it to the size the kernel uses. I thought it would be better to have a sane default in se->bufsize at all times.